pkg/acquisition/modules/appsec/appsec_runner.go

```diff
@@ -35,19 +35,24 @@ type AppsecRunner struct {
 func (r *AppsecRunner) MergeDedupRules(collections []appsec.AppsecCollection, logger *log.Entry) string {
 	var rulesArr []string
 	dedupRules := make(map[string]struct{})
+	discarded := 0
 
 	for _, collection := range collections {
+		// Dedup *our* rules
 		for _, rule := range collection.Rules {
 			if _, ok := dedupRules[rule]; !ok {
 				rulesArr = append(rulesArr, rule)
 				dedupRules[rule] = struct{}{}
 			} else {
+				discarded++
 				logger.Debugf("Discarding duplicate rule : %s", rule)
 			}
 		}
+		// Don't mess up with native modsec rules
+		rulesArr = append(rulesArr, collection.NativeRules...)
 	}
-	if len(rulesArr) != len(dedupRules) {
-		logger.Warningf("%d rules were discarded as they were duplicates", len(rulesArr)-len(dedupRules))
+	if discarded > 0 {
```

Comment:
not an issue but personally I would have done

```go
if discarded := len(rulesArr) - len(dedupRules); discarded > 0 {
  logger.Warningf("%d rules were discarded as they were duplicates", discarded)
}
```

Reply:
actually I see what manu was saying beforehand so yeah we need to track an extra variable, classing as resolved.